### PR TITLE
fix(native-filters): Default value not shown in native filters

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY
When user created a new filter with a default value, that value wasn't displayed in the filter in filters panel, but could be applied to charts. This PR fixes this issue.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:

https://user-images.githubusercontent.com/15073128/155991441-b4c1291c-a701-4d4b-9ac0-49bc4a863f86.mov

After:

https://user-images.githubusercontent.com/15073128/155991234-b1e81404-4812-4f7f-833a-4fefe1a9111e.mov

### TESTING INSTRUCTIONS
1. Create a native filter with default value and save
2. Verify that selected default value is displayed in the filter in filters panel
3. Reload the page
4. Verify that after reload the default value is still applied and visible in filter

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/dashboard/components/nativeFilters/FilterBar/index.tsx [hunk 1]
```diff
@@ -288,6 +288,10 @@ const FilterBar: React.FC<FiltersBarProps> = ({
     }
   }, [JSON.stringify(filters), JSON.stringify(previousFilters)]);
 
+  useEffect(() => {
+    setDataMaskSelected(() => dataMaskApplied);
+  }, [JSON.stringify(dataMaskApplied), setDataMaskSelected]);
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/nativeFilters/FilterBar/index.tsx:293]: Can you use `dataMaskAppliedText`? It's already converting the JSON to string.

Comment (kgabryje) [reply superset-frontend/src/dashboard/components/nativeFilters/FilterBar/index.tsx]: Good point! Done

**Comment by jinghua-qa** [comment]
I found an issue that in number filter, when user choose minimum or maximum for single value, the default chosen value will be off

https://user-images.githubusercontent.com/81597121/156037299-dec314c6-ff73-4b9b-abff-dc630805dc56.mov

**Comment by michael-s-molina** [comment]
@jinghua-qa @kgabryje Let's tackle the number filter problem in another PR since this is already happening on master.

**Review by michael-s-molina — APPROVED** [review]
LGTM.

**Comment by sadpandajoe** [comment]
🏷️  preset:2022.9

**Pull request merged**